fix: ensure reserved keywords can never be rendered as properties for Java class

jonaslagoni opened the pull request:
**Description**
This PR ensures that reserved keywords can never be rendered for the Java class.

I actually struggled quite a bit with how this functionality should work in cohesion with the name convention option. I could not come up with any other way of solving this problem other than this way. In case you see another please let me know 🙏 

**Related issue(s)**
partly solves #176

jonaslagoni commented:
Blocked by https://github.com/asyncapi/modelina/pull/266

jonaslagoni commented:
Once we agree that this approach is as expected I will replicate it across all generators.

jonaslagoni commented on docs/generators.md:
```diff
@@ -24,6 +24,7 @@ Default options contain:
 | Option | Type | Description | Default value |
 |---|---|---|---|
 | `indentation` | Object | Options for indentation. | - |
+| `reservedPropertyWord` | String | Word or character to append to property name when reserved keywords or duplicate property names are encountered. | 'reserved' |
```
Hmm, yea you are right 🤔 Good suggestion 👍

magicmatatjahu commented on src/models/Preset.ts:
```diff
@@ -24,7 +24,8 @@ export enum PropertyType {
   property: CommonModel;
-  type: PropertyType
+  type: PropertyType;
+  modelPropertyName: string;
```
For what you need `modelPropertyName`? As I see in the implementation, it's the same as `propertyName` and also you don't use it anywhere.

jonaslagoni replied:
Good catch, left over from old implementation, removing!

magicmatatjahu commented on src/helpers/NameHelpers.ts:
```diff
@@ -19,29 +19,47 @@ export enum DefaultPropertyNames {
+      // If name is considered reserved, make sure we rename it appropriately
+      // and make sure no clashes occur.
+      name = FormatHelpers.toCamelCase(`reserved_${name}`);
+      if (Object.keys(ctx.model.properties || {}).includes(name)) {
```
Good catch! 💪🏼

magicmatatjahu reviewed (changes requested):
Good work! Please refer to my comments, and also please add missed TS tests :)

jonaslagoni commented:
> Good work! Please refer to my comments, and also please add missed TS tests :)

I just added the keyword list, next PR is gonna be to add the implementation of using the list the same way Java has been adapted. Just didnt want to remove it now that I had added it 😅

magicmatatjahu commented:
@jonaslagoni Ok, remember about promise 😄

magicmatatjahu reviewed (approved):
LGTM!

jonaslagoni commented:
Thanks 🙇
